src/posix/platform/spinel_interface.hpp

```diff
@@ -0,0 +1,180 @@
+        kMaxFrameSize = 2048, ///< Maximum frame size (number of bytes).
+    };
+
+    /**
+     * This type defines a receive frame buffer to store received spinel frame(s).
+     *
+     * @note The receive frame buffer is an `Hdlc::MultiFrameBuffer` and therefore it is capable of storing multiple
+     * frames in a FIFO queue manner.
+     *
+     */
+    typedef Hdlc::MultiFrameBuffer<kMaxFrameSize> RxFrameBuffer;
+
+    /**
+     * This class defines the callbacks provided by `SpinelInterface` to its owner/user.
+     *
+     */
+    class Callbacks
+    {
+    public:
+        /**
+         * This callback is invoked to notify owner/user of `SpinelInterface` of a received spinel frame.
+         *
+         * The newly received frame is available in `RxFrameBuffer` from `SpinelInterface::GetRxFrameBuffer()`.
+         * User can read and process the frame. The callback is expected to either discard the new frame using
+         * `RxFrameBuffer::DiscardFrame()` or save the frame using `RxFrameBuffer::SaveFrame()` to be read and
+         * processed later.
+         *
+         */
+        virtual void HandleReceivedFrame(void) = 0;
+    };
+
+    /**
+     * This constructor initializes the object.
+     *
+     */
+    SpinelInterface()
+        : mRxFrameBuffer()
+    {
+    }
+
+    /**
+     * This method gets the `RxFrameBuffer`.
+     *
+     * The receive frame buffer is an `Hdlc::MultiFrameBuffer` and therefore it is capable of storing multiple
+     * spinel frames in a FIFO queue manner. The `RxFrameBuffer` contains the received spinel frames.
+     *
+     * When during `Process()` or `WaitForFrame()` the `Callbacks::HandleReceivedFrame()` is invoked, the newly
+     * received spinel frame is available in the receive frame buffer. The callback is expected to either process
+     * and then discard the frame (using `RxFrameBuffer::DiscardFrame()` method) or save the frame
+     * (using `RxFrameBuffer::SaveFrame()` so that it can be read later.
+     *
+     * @returns A reference to receive frame buffer containing newly received spinel frame or previously saved spinel
+     *          frames.
+     *
+     */
+    RxFrameBuffer &GetRxFrameBuffer(void) { return mRxFrameBuffer; }
+
+    /**
+     * This method initializes the interface to the Radio Co-processor (RCP)
+     *
+     * @note This method should be called before reading and sending spinel frames to the interface.
+     *
+     * @param[in]   aConfig  Parameters to be given to the device or executable.
+     *
+     * @retval OT_ERROR_NONE          The interface is initialized successfully
+     * @retval OT_ERROR_ALREADY       The interface is already initialized.
+     * @retval OT_ERROR_INVALID_ARGS  The device or executable cannot be found or failed to open/run.
+     *
+     */
+    virtual otError Init(const char *aRadioFile, const char *aRadioConfig) = 0;
```

Comment:
Missing doxygen for `aRadioFile` and `aRadioConfig`.

Reply:
Updated.